.github/workflows/sphinx_docs.yaml

```diff
@@ -0,0 +1,35 @@
+name: Build API Reference docs using sphinx 
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
It would make sense to have this workflow run on PR as well - as a test that sphinx is working and happy.